test/test_time_series.jl

```diff
@@ -332,13 +346,306 @@ end
+    ) == TimeSeries.values(data)
+    @test IS.get_time_series_timestamps(
+        IS.SingleTimeSeries,
+        component,
+        ts_name;
+        scenario = "low",
+        model_year = "2030",
+    ) == TimeSeries.timestamp(data)
+
+    IS.add_time_series!(sys, component, ts; scenario = "high", model_year = "2030")
+    IS.add_time_series!(sys, component, ts; scenario = "low", model_year = "2035")
+    IS.add_time_series!(sys, component, ts; scenario = "high", model_year = "2035")
+
+    @test_throws ArgumentError IS.get_time_series(
+        IS.SingleTimeSeries,
+        component,
+        ts_name,
+    )
+    @test_throws ArgumentError IS.get_time_series(
+        IS.SingleTimeSeries,
+        component,
+        ts_name,
+        scenario = "low",
+    )
+    @test IS.get_time_series(
+        IS.SingleTimeSeries,
+        component,
+        ts_name;
+        scenario = "low",
+        model_year = "2035",
+    ) isa IS.SingleTimeSeries
+    @test IS.has_time_series(component, IS.SingleTimeSeries)
+    @test IS.has_time_series(component, IS.SingleTimeSeries, ts_name)
+    @test IS.has_time_series(component, IS.SingleTimeSeries, ts_name, scenario = "low")
+    @test IS.has_time_series(
+        component,
+        IS.SingleTimeSeries,
+        ts_name,
+        model_year = "2030",
+        scenario = "low",
+    )
+    @test IS.has_time_series(
+        component,
+        IS.SingleTimeSeries,
+        ts_name,
+        model_year = "2030",
+        scenario = "low",
+    )
+    @test !IS.has_time_series(
+        component,
+        IS.SingleTimeSeries,
+        ts_name;
+        model_year = "2060",
+        scenario = "low",
+    )
+    @test length(IS.list_time_series_info(component)) == 4
+    @test IS.list_time_series_info(component)[1].type === IS.SingleTimeSeries
+    @test Tables.rowtable(
+        IS.sql(
+            sys.time_series_manager.metadata_store,
+            "SELECT COUNT(*) AS count FROM $(IS.METADATA_TABLE_NAME)",
+        ),
+    )[1].count == 4
+end
+
+@testset "Test add with features with mixed types" begin
+    sys = IS.SystemData()
+    name = "Component1"
+    component = IS.TestComponent(name, 5)
+    IS.add_component!(sys, component)
+
+    initial_time = Dates.DateTime("2020-09-01")
+    resolution = Dates.Hour(1)
+
+    data = TimeSeries.TimeArray(
+        range(initial_time; length = 365, step = resolution),
+        rand(365),
+    )
+    ts_name = "test"
+    ts = IS.SingleTimeSeries(; data = data, name = ts_name)
+    IS.add_time_series!(sys, component, ts; scenario = "low", model_year = "2030")
+    @test IS.has_time_series(
+        component,
+        IS.SingleTimeSeries,
+        ts_name;
+        scenario = "low",
+        model_year = "2030",
+    )
+    @test !IS.has_time_series(
+        component,
+        IS.SingleTimeSeries,
+        ts_name;
+        scenario = "low",
+        model_year = 2030,
+    )
+    IS.add_time_series!(sys, component, ts; scenario = "low", model_year = 2030)
```

Comment:
Since we're already checking for duplicates when adding time series, seems like checking the types wouldn't add too much more query, right?

One reason someone might want mixed types is if they have an integer feature but want a null value.

Reply:
The check for duplicates is a fast check because it uses indexed columns. Feature types are not in a column. We would have to compute the types for specific feature names on every insertion. Or keep a local cache of what we've seen. And we would have to load those at deserialization. I'm not sure the complexity is worth it.

Reply:
Okay. Let's keep it as-is for now, I'll add this to the deferred issues.